qiita_db/handlers/plugin.py

```diff
@@ -0,0 +1,186 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from json import loads, dumps
+
+from tornado.web import HTTPError
+
+from .oauth2 import OauthBaseHandler, authenticate_oauth
+import qiita_db as qdb
+
+
+def _get_plugin(name, version):
+    """Returns the plugin with the given name and version
+
+    Parameters
+    ----------
+    name : str
+        The name of the plugin
+    version : str
+        The version of the plugin
+
+    Returns
+    -------
+    qiita_db.software.Software
+        The requested plugin
+
+    Raises
+    ------
+    HTTPError
+        If the plugin does not exist, with error code 404
+        If there is a problem instantiating the plugin, with error code 500
+    """
+    try:
+        plugin = qdb.software.Software.from_name_and_version(name, version)
+    except qdb.exceptions.QiitaDBUnknownIDError:
+        raise HTTPError(404)
+    except Exception as e:
```

Comment:
If it's not possible to generate them, why test them? 

Now, this is working on `from_name_and_version`, which only raises `QiitaDBUnknownIDError`, so the second except is working on:
```
            sql = """SELECT software_id
                     FROM qiita.software
                     WHERE name = %s AND version = %s"""
            qdb.sql_connection.TRN.add(sql, [name, version])
            res = qdb.sql_connection.TRN.execute_fetchindex()
```
which will only fail if the DB fails or the structure changes, right? If that's correct I think we can remove them, right?